### DISCUSSION TITLE
Script problem after RPI5 update and NVME upgrade

### DISCUSSION THREAD

**Discussion by ldb20151** [aaronwmorris/indi-allsky#1698] [Q&A] [answered]
I previously asked question "Is my GPS working? #1634" ... this question is somewhat related but not the same issue.
While working through the GPS activation in 1634, I decided to upgrade the NMVE SSD card on my RPI 5.  I had AllSky Camera working at a basic level -- without the GPS completed -- before swapping the NVME card.  I imaged the NMVE card onto a larger card, installed the new card into the RPI and was able to successfully boot to the new card and access the AllSky camera.  So far, so good.  
At this point, the RPI asked to do updates to a wide variety of software packages.  I allowed it to complete all upgrades.  Perhaps this was a mistake, but being worried about security of the system I assumed this was the thing to do.  Once updates were completed and system rebooted, I'm still able to access AllSky camera, but when I attempt to press ahead with GPS activation as discussed in the WIKI and the above GPS thread, I ran into a problem  When I try to run the GPS script from command line,   
 ./misc/setup_gpsd.sh    
the pi returns an error message file not found.  It appears there's no such folder.
I had to put this project aside for a couple weeks so at this point I'm not 100% sure it worked the first time I set it up before the NVME swap -- I know I did not complete checkout of the GPS functions.  However, its definitely not working now.  
Per my original post above, I'm a complete pi & linux newby so likely I'm just missing something obvious.  Can anyone give me a clue how to diagnose this?  Is the referenced script part of the original install package or do I need to fetch it from somewhere?  If it's missing from original install, any suggestions on how that could happen or how to fix it?  I used a disc imaging program (EASE) to image the drive (single partition) and checked it before installing in in the RPI and it looked ok.   Everything else seems to work OK. 
My GPS is the standard Adafruit GPS hat -- folks seem to have good experience with it once working.
Thanks in advance for any help you can provide.  I hope to have time to focus on this over the holidays.

**Comment by aaronwmorris** [comment]
The setup_gpsd.sh script is failing because it assumes /dev/ttyACM0 which is very common for USB GPS adapters.  You can specify the device name as part of the setup_gpsd.sh command line arguments.

First you will need to run `sudo raspi-config` and enable the serial port per the Adafruit guide.

Then you can run the setup_gpsd.sh script with the serial port name.

```
./misc/setup_gpsd.sh /dev/serial0
```

**Reply by ldb20151** [reply]
Thanks again for your help.  Your note sent me back to the Adafruit manual and their quick test procedures.  Turns out, my unit requires ttyAMA0 vs serial0 or ttyACM0.  The manual is very confusing.  It "says" the V3 Ultimate hat (that's what I have) should use serial0 but it will absolutely not work in their test process.  I found ttyAMA0 listed in an inset example screenshot GPS output -- tried it and it works great.  Passes the Adafruit manual's raw data flow test and works great in the GPSd script as well.  I can't figure out how to screen shot the command window (gnome doesn't like it) but it has my location pretty close I think -- time looks good as well.  Its impressive performance since I'm working in a basement workshop without clear view of the sky.  It has managed to lock on to 8 satellites and can see 12 -- although some of the 8 wink out from time to time.  I'm using the external antenna but its laying on the bench beside me, so...
At first, I didn't believe the data but when I turn the antenna over it loses all satellites and the red light on the hat starts to flash rapidly which means loss of lock.  So, its acquiring something.  Skies are clear today but there's a lot of wood above me.  I'll keep checking for accuracy but I think GPS is working.

Just to be sure I understand what the script did/didn't do, **should I now proceed with the time sync as described in the wiki:  sudo apt-get install ntpsec**
I stopped short of this since, as a newby, I didn't want to reload a bunch of stuff that might already be done.  Not clear to me if that would garbage things up of if the manager would just handle it.

**Reply by aaronwmorris** [reply]
Excellent!  The setup_gpsd.sh script is certainly not comprehensive, but just enough to get the gpsd subsystem activate.

You can certainly move on to ntpsec now.  The gps setup script only establishes the GPS setup for location services.  I did not really want to touch the NTP config since people may have that already customized.

One interesting thing you can look into is since you have a GPS HAT, I believe it is also setup to send the PPS signal to a pin on the raspberry pi.  PPS is used when you want sub-millisecond time accuracy.  Most people do not need it, but it is a neat feature.

**Reply by ldb20151** [reply, marked as answer]
I answered my own question -- I do need to press ahead with the NTP sync and am working in now.  Thx.

**Reply by ldb20151** [reply]
Yes, the ultimate manual does mention the PPS pin.  I have been thinking about how I might use it -- hardware is more my thing than software, which is killing me right now.  As I understand most pi GPS's (the ones offered by Adafruit anyway) can output a 'pulse per second' for synchronizing the time. There's an available breakout for this plus a closed jumper that connects it to GPIO pin#4.  The book says if you do not need PPS and want to  use #4 for something else, you must cut the little connection on the PPS PCB line.  If you forget to do this, GPIO pin 4 will be unavailable for any other use.  I'm also looking at GPIO control of basic environmental controls and maybe some filters or focuser in the future so knowing how all these pins are being used will be important.   It would be nice to have a config.file to disable the PPS as well but that doesn't seem an option.  For now, I'll likely put a removable jumper on that line unless I can find a use for it.  Until I get the baseline system up and running (and properly tuned) I don't want to make modifications.
Ultimate GPS Hat Reference can be found here:  https://cdn-learn.adafruit.com/downloads/pdf/adafruit-ultimate-gps-hat-for-raspberry-pi.pdf
Link to much more detailed Adafruit GPS Hat Tech Manual can be found on page 5 of that .pdf for those who might be interested.  Very interesting stuff for those into GPS.
